ironforgedbot/tasks/ranks.py

```diff
@@ -0,0 +1,82 @@
+import random
+import time
+
+import asyncio
+import discord
+
+from ironforgedbot.commands.hiscore.calculator import points_total
+from ironforgedbot.common.helpers import normalize_discord_string
+from ironforgedbot.common.ranks import RANKS, get_rank_from_points
+
+
+def refresh_ranks(guild: discord.Guild, updates_channel_name: str, loop: asyncio.BaseEventLoop):
+    if updates_channel_name is None or "" == updates_channel_name:
+        return
+
+    updates_channel = _find_channel(guild, updates_channel_name)
+    if updates_channel is None:
+        return
+
+    icons = _load_icons(guild)
+    members_to_update = {}
+
+    # Just in case we don't want to handle routing from guild.members
+    for member in guild.members:
+        if member.bot or member.nick is None or "" == member.nick:
+            continue
+
+        nick = normalize_discord_string(member.nick)
+        if "" == nick:
+            continue
+
+        current_role = _find_rank(member)
+        if current_role is None:
+            continue
+
+        members_to_update[nick] = current_role
+
+    for member, current_role in members_to_update.items():
+        try:
+            current_points = points_total(member)
+        except RuntimeError:
+            continue
+
+        actual_role = get_rank_from_points(current_points)
+        if current_role != str(actual_role):
+            message = (f'{member} has upgraded their rank from {icons[current_role]} to {icons[actual_role]} '
+                       f'with {current_points} points')
+            asyncio.run_coroutine_threadsafe(_send_message(updates_channel, message), loop)
+
+        time.sleep(random.randint(1, 5))
```

Comment:
comment: what is the purpose of this randomized sleep here?

Reply:
Found no info about jagex api rates and usage policies. We have 400+ members plus random /score calls. Since this is a background job, we don't care how long it runs, figured I'll disperse the API call a bit for some extra safety.

Reply:
Makes sense, thanks for the clarification!